File: src/15/15051.cpp
/**
 * 15051. Máquina de café
 * 
 * 작성자: Raccoon Dog
 * 언어: C++14
 * 사용 메모리: 1,984 KB
 * 소요 시간: 0 ms
 * 해결 날짜: 2020년 9월 20일
 */

#include <iostream>
#include <cstdlib>

using namespace std;

int main()
{
    int n[3] = { }, times = 0, mintime = 0;

    cin >> n[0] >> n[1] >> n[2];

    for (int i = 0; i < 3; i++)
    {
        for (int j = 0; j < 3; j++)
        {
            times += (n[j] * abs(j - i) * 2);
        } 

        if (i == 0)
        { 
            mintime = times;
        }
        else if (times < mintime)
        { 
            mintime = times;
        }

        times = 0;
    }

    cout << mintime;

    return 0;
}

File: src/15/15051.py
"""
15051. Máquina de café

작성자: xCrypt0r
언어: Python 3
사용 메모리: 29,380 KB
소요 시간: 64 ms
해결 날짜: 2020년 9월 20일
"""

def check_time(floor, people):
    time = 0

    for i in range(3):
        time += abs(floor - i) * people[i] * 2

    return time

def main():
    A = [int(input()) for _ in range(3)]
    time = [check_time(i, A) for i in range(3)]

    print(min(time))

if __name__ == '__main__':
    main()
